plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -553,4 +557,59 @@ protected Session joinPushdownEnabled(Session session)
                 .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                 .build();
     }
+
+    @Test(dataProvider = "testCaseInsensitivePredicatePushdownDataProvider")
+    public void testCaseInsensitivePredicatePushdown(String type)
+    {
+        Session useCaseInsensitiveVarcharPushdown = Session.builder(getSession())
+                .setCatalogSessionProperty(getSession().getCatalog().orElseThrow(), UNSAFE_VARCHAR_PUSHDOWN, "true")
+                .build();
+
+        if (hasBehavior(SUPPORTS_PREDICATE_PUSHDOWN_WITH_VARCHAR_INEQUALITY)) {
+            assertQueryFails(useCaseInsensitiveVarcharPushdown, "SHOW SESSION", "Unknown session property .*.unsafe-varchar-pushdown");
+            return;
+        }
+        String tableName = "test_case_insensitive_predicate_pushdown" + randomTableSuffix();
+        assertQuerySucceeds(format("CREATE TABLE %s (value %s)", tableName, type));
+        assertQuerySucceeds(format("INSERT INTO %s VALUES 'a', 'A', 'b', 'B'", tableName));
+
+        assertThat(query(useCaseInsensitiveVarcharPushdown, format("SELECT value FROM %s WHERE value = 'a'", tableName)))
+                // possibly incorrect results with unexpected 'A'
+                .isFullyPushedDown();
```

Comment:
I don't understand. `isFullyPushedDown` verifies results correctness, this shouldn't pass.

Reply:
That is because `allow_pushdown_into_connectors` does not control predicate pushdown.